### PR TITLE
fix: making all repository generation private per default

### LINKED ISSUE #5186: jx import creates git repos as public, they should be private by default instead
### Summary
We should assume repos should be private by default else code and secrets could be accidentally leaked.
### Expected behaviour
When importing a project, create Git repo as PRIVATE
When creating a new cluster environment repos are PRIVATE
### Actual behaviour
When importing a project, Git repo is PUBLIC and not unsecured.

### PR DESCRIPTION
**Pull request by hferentschik**
Fixes #5186, #5341, #5476, #5472

Following up on #5511 which reverted the defaulting to private repos

### CODE REVIEW

**Comment by hferentschik** [comment]
will need a fix for #5515 first.

File: pkg/cmd/deletecmd/delete_application.go [hunk 1]
```diff
@@ -481,16 +482,18 @@ func (o *DeleteApplicationOptions) init() error {
 	return nil
 }
 
-func (o *DeleteApplicationOptions) deletePipelineActivitiesForSourceRepository(jxClient versioned.Interface, ns string, srName string) error {
+func (o *DeleteApplicationOptions) deletePipelineActivitiesForSourceRepository(jxClient versioned.Interface, ns string, gitOwner string, gitRepo string) error {
+	gitOwnerSelector := fields.OneTermEqualSelector("metadata.spec.gitOwner", gitOwner)
```
Comment (hferentschik) [inline pkg/cmd/deletecmd/delete_application.go]: @abayer see whether this is going to cut it. If this gets us past the error with labels in the BDD tests, I can create a dedicated PR for #5515

**Comment by hferentschik** [comment]
@abayer, so your fix for the BDD command line as well as the changes I added for removing the sourcerepository label seem to work. I verified that the created test repos were private:

```
jenkins-x-bot-test/environment-pr-5523-20-tekton-gkebdd-production created 2019-09-18 07:24:50 UTC private true
jenkins-x-bot-test/environment-pr-5523-20-tekton-gkebdd-staging created 2019-09-18 07:24:42 UTC private true
```

**Comment by hferentschik** [comment]
@abayer now rebased on #5526. Is there anything else you think we should address before we can get this back in?

**Comment by abayer** [comment]
Looks reasonable to me - the tests are using private repos, so if the tekton tests pass, we're good.

**Pull request merged**